src/components/utils/src/logger/logger_impl.cc

```diff
@@ -71,9 +73,16 @@ void LoggerImpl::DeInit() {
 }
 
 void LoggerImpl::Flush() {
-  if (use_message_loop_thread_) {
-    if (loop_thread_) {
+  if (use_message_loop_thread_ && loop_thread_) {
+    if (0 == flush_logs_time_point_.time_since_epoch().count()) {
       loop_thread_->WaitDumpQueue();
+    } else if (settings_->flush_log_messages_before_shutdown()) {
+      // Calculate elapsed seconds since IGNITION_OFF
+      auto seconds = std::chrono::duration_cast<std::chrono::seconds>(
+          std::chrono::high_resolution_clock::now() - flush_logs_time_point_);
+      uint16_t seconds_before_shutdown =
```

Comment:
@VladSemenyuk I recommend using signed int here as there is a probability that this number may be negative (ex. if some transport is closing more than 30 sec) so you should skip timed wait in that case.

Reply:
Done